src/randoop/main/GenTests.java

```diff
@@ -429,264 +399,335 @@ public boolean handle(String[] args) throws RandoopTextuiException {
+   */
+  private void outputCoverageMap(CodeCoverageTracker covTracker) throws Error {
+    try {
+      FileOutputStream fileos = new FileOutputStream(output_covmap);
+      ObjectOutputStream objectos = new ObjectOutputStream(new GZIPOutputStream(fileos));
+      objectos.writeObject(covTracker.branchesToCoveringSeqs);
+      objectos.close();
+      fileos.close();
+    } catch (Exception e) {
+      throw new Error(e);
     }
+  }
 
-    List<File> genfiles = write_junit_tests (junit_output_dir, sequences,
-        pluginBridge == null ? null : pluginBridge.additionalJunitClasses);
 
-    if (pluginBridge != null) {
-      for (File f : genfiles) {
-        // TODO: This is unsafe. I'm just assuming that the list file written is the driver file
-        boolean isDriver = genfiles.lastIndexOf(f) == genfiles.size() - 1;
-        pluginBridge.msgSender.send(new CreatedJUnitFile(f, isDriver));
+  private void outputCoverageBranches(CodeCoverageTracker covTracker) throws Error {
+    Comparator<Branch> branchComparator = new Comparator<Branch>() {
+      public int compare(Branch o1, Branch o2) {
+        return o1.toString().compareTo(o2.toString());
       }
+    };
+    Set<Branch> branches = new TreeSet<Branch>(branchComparator);
+    branches.addAll(covTracker.branchesCovered);
+    // Create a file with branches, sorted by their string representation.
+    BufferedWriter writer = null;
+    try {
+      writer = new BufferedWriter(new FileWriter(output_branches));
+      // Touch all covered branches (they may have been reset during generation).
+      for (Branch b : branches) {
+        writer.append(b.toString());
+        writer.newLine();
+      }
+      writer.close();
+    } catch (IOException e) {
+      throw new Error(e);
     }
+  }
 
+  /**
+   * Returns the list of contracts to be used in contract checking.
+   * 
+   * @param classes  the list of annotated classes for retrieving contracts
+   * @return the list of {@code ObjectContract} objects for contract checking
+   */
+  private List<ObjectContract> getContracts(List<Class<?>> classes) {
+    List<ObjectContract> contracts = new ArrayList<ObjectContract>();
+
+    // Add any @CheckRep-annotated methods and create visitors for them.
+    List<ObjectContract> checkRepContracts = getContractsFromAnnotations(classes);
+    contracts.addAll(checkRepContracts);
+
+    // Now add all of Randoop's default contracts.
+    // Note: if you add to this list, also update the Javadoc for check_object_contracts.
+    contracts.add(new EqualsReflexive());
+    contracts.add(new EqualsSymmetric());
+    contracts.add(new EqualsHashcode());
+    contracts.add(new EqualsToNullRetFalse());
+    return contracts;
+  }
 
-
-    if (pluginBridge != null) {
-      IMessage msg = new ClosingStream();
-      pluginBridge.msgSender.send(msg);
+  /**
+   * Constructs an {@code ExceptionPredicate} that checks whether each kind of
```

Comment:
I changed the text to try to make it more clear.

As far as the independent predicates is concerned, there is a problem, but it is opposite of what you are saying.  Exceptions are classified by checking for behaviors in the order: invalid, error, expected. So, the problem is if the more general case is invalid/error and the more specific is expected.  I added the checks for this case to the constructed predicate. The construction assumes the enum constant order is invalid < error < expected, and this is noted in comments in the enum.